db/migrate/20170601161315_add_anonymous_user_to_posts_without_userid.rb

```diff
@@ -0,0 +1,20 @@
+
+class AddAnonymousUserToPostsWithoutUserid < ActiveRecord::Migration[5.1]
+  def up
+    password = (0...15).map { ('a'..'z').to_a[rand(25)] }.join
+    user = User.create(name: "Anonymous", email: "anon@acebook.com", password: password)
+    Post.all.each { |post|
+      post.user_id ||= user.id
+      post.save
+    }
+  end
+
+  def down
+    user = User.find_by(email: "anon@acebook.com")
+    user.posts.each { |post|
+      post.user_id = nil
+      post.save(validate: false)
+    }
+    user.destroy
```

Comment:
Same here for the `do/end` and `destroy` -> `destroy!`, `save` -> `save!`, `find_by` -> `find_by!`